ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDTopicTableViewController.m

```diff
@@ -0,0 +1,223 @@
+
+- (void)controllerWillChangeContent:(NSFetchedResultsController *)controller {
+	
+	[self.tableView beginUpdates];
+	
+}
+
+- (void)controller:(NSFetchedResultsController *)controller didChangeObject:(id)anObject
+	   atIndexPath:(NSIndexPath *)indexPath forChangeType:(NSFetchedResultsChangeType)type
+	  newIndexPath:(NSIndexPath *)newIndexPath {
+	
+	switch (type) {
+		case NSFetchedResultsChangeInsert:
+			[self.tableView insertRowsAtIndexPaths:[NSArray arrayWithObject:newIndexPath]
+								  withRowAnimation:UITableViewRowAnimationFade];
+			break;
+		case NSFetchedResultsChangeDelete:
+			[self.tableView deleteRowsAtIndexPaths:[NSArray arrayWithObject:indexPath]
+								  withRowAnimation:UITableViewRowAnimationFade];
+			break;
+		case NSFetchedResultsChangeUpdate: {
+			UITableViewCell *cell = [self.tableView cellForRowAtIndexPath:indexPath];
+			CDTopic *topicItem = [self.fetchedResultsController objectAtIndexPath:indexPath];
+			cell.textLabel.text = topicItem.title;
+		}
+			break;
+		case NSFetchedResultsChangeMove:
+			[self.tableView deleteRowsAtIndexPaths:[NSArray arrayWithObject:indexPath]
+								  withRowAnimation:UITableViewRowAnimationFade];
+			[self.tableView insertRowsAtIndexPaths:[NSArray arrayWithObject:newIndexPath]
+								  withRowAnimation:UITableViewRowAnimationFade];
+			break;
+		default:
+			break;
+	}
+	
+}
+
+- (void)controllerDidChangeContent:(NSFetchedResultsController *)controller {
+	
+	[self.tableView endUpdates];
+	
+}
+
+#pragma mark - DataSource and Delegate
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *) tableView {
+	
+	return self.fetchedResultsController.sections.count;
+
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+
+	return self.fetchedResultsController.sections[section].numberOfObjects;
+
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+	
+	UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"SimpleReminderCell"];
+	
+	if (cell) {
+		CDTopic *topic = [self.fetchedResultsController objectAtIndexPath:indexPath];
+	
+		cell.textLabel.text = topic.title;
+	}
+	
+	return cell;
+	
+}
+
+- (void)tableView:(UITableView *)tableView moveRowAtIndexPath:(NSIndexPath *)sourceIndexPath
+	  toIndexPath:(NSIndexPath *)destinationIndexPath {
+	
+	NSLog(@"MovedRow from row %ld, to row %ld", (long)sourceIndexPath.row, (long)destinationIndexPath.row);
+	
+}
+
+- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle
+forRowAtIndexPath:(NSIndexPath *)indexPath {
+	
+	if (editingStyle == UITableViewCellEditingStyleDelete) {
+		CDTopic *topicToDelete = [self.fetchedResultsController objectAtIndexPath:indexPath];
+		[self.managedContext deleteObject:topicToDelete];
+		
+		NSError *error;
+		if (![self.managedContext save:&error]) {
+			NSLog(@"An error occured: %@", error.localizedDescription);
```

Comment:
error.localizedDescription e mai mult pentru alerte sau ce sa prezinti la user pe UI. Cand logezi o eroare pune obiectul "error" ca vrei sa ai in loguri cat mai multe informatii(codul, domeniul erorii ...).